app/src/main/java/org/literacyapp/authentication/AuthenticationActivity.java

```diff
@@ -0,0 +1,179 @@
+import ch.zhaw.facerecognitionlibrary.Recognition.SupportVectorMachine;
+import ch.zhaw.facerecognitionlibrary.Recognition.TensorFlow;
+
+public class AuthenticationActivity extends AppCompatActivity implements CameraBridgeViewBase.CvCameraViewListener2 {
+    private static final int NUMBER_OF_MAXIMUM_TRIES = 5;
+    private SupportVectorMachine svm;
+    private TensorFlow tensorFlow;
+    private PreProcessorFactory ppF;
+    private JavaCameraView preview;
+    private AnimalOverlayHelper animalOverlayHelper;
+    private StudentImageCollectionEventDao studentImageCollectionEventDao;
+    private int numberOfTries;
+
+    static {
+        if (!OpenCVLoader.initDebug()) {
+            // Handle initialization error
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_authentication);
+
+        MediaPlayer mediaPlayer = MediaPlayer.create(this, R.raw.face_instruction);
+        mediaPlayer.start();
+
+        preview = (JavaCameraView) findViewById(R.id.CameraView);
+
+        // Use front-camera
+        preview.setCameraIndex(1);
+
+        preview.setVisibility(SurfaceView.VISIBLE);
+        preview.setCvCameraViewListener(this);
+
+        animalOverlayHelper = new AnimalOverlayHelper(getApplicationContext());
+
+        // Initialize DB Session
+        LiteracyApplication literacyApplication = (LiteracyApplication) getApplicationContext();
+        DaoSession daoSession = literacyApplication.getDaoSession();
+        studentImageCollectionEventDao = daoSession.getStudentImageCollectionEventDao();
+    }
+
+    @Override
+    public void onCameraViewStarted(int width, int height) {
+
+    }
+
+    @Override
+    public void onCameraViewStopped() {
+
+    }
+
+    @Override
+    public Mat onCameraFrame(CameraBridgeViewBase.CvCameraViewFrame inputFrame) {
+        Mat imgRgba = inputFrame.rgba();
+        Mat imgCopy = new Mat();
+
+        // Store original image for face recognition
+        imgRgba.copyTo(imgCopy);
+
+        // Mirror front camera image
+        Core.flip(imgRgba,imgRgba,1);
+
+        List<Mat> images = ppF.getCroppedImage(imgCopy);
+        if (images != null && images.size() == 1){
+            Mat img = images.get(0);
+            if (img != null){
+                Rect[] faces = ppF.getFacesForRecognition();
+                if (faces != null && faces.length == 1){
+                    String svmString = getSvmString(img);
+                    String svmProbability = svm.recognizeProbability(svmString);
+                    Student student = getStudentFromProbability(svmProbability);
+                    numberOfTries++;
+                    if (student != null){
+                        new StudentUpdateHelper(getApplicationContext(), student).updateStudent();
+                        finish();
+                    } else if (numberOfTries >= NUMBER_OF_MAXIMUM_TRIES){
```

Comment:
I don't think so. numberOfTries starts with 0 and increases until 5 and then the StudentImageCollectionActivity would be triggered because the recognition probability was never above 90%.